CMakeLists.txt

```diff
@@ -17,22 +18,45 @@ option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
+set(BOARD_NAME "" CACHE INTERNAL "Enter the board name to compile mlpack for.")
```

Comment:
I worried that this would override any value of `BOARD_NAME` that had been set, but then I found this in the CMake documentation:

> Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default.

So, I guess this is just fine. :)